activerecord/test/cases/migration/references_foreign_key_test.rb

```diff
@@ -247,6 +247,17 @@ def test_references_foreign_key_with_suffix
           fk_definitions = fks.map { |fk| [fk.from_table, fk.to_table, fk.column] }
           assert_equal([["testings", "testing_parents", "parent2_id"]], fk_definitions)
         end
+
+        if current_adapter?(:PostgreSQLAdapter)
+          test "add_reference combines adding column and foreign key" do
+            @connection.create_table :testings
+            assert_queries(1) do
```

Comment:
If I understand it well, it was one query before as well. Should we check more on query itself for proper format to be used?

Reply:
It was 2 queries before: https://github.com/rails/rails/blob/a4662721965e127bfd7de2192dbd829339ab2058/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb#L976-L978
https://github.com/rails/rails/blob/a4662721965e127bfd7de2192dbd829339ab2058/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb#L187-L199

Reply:
Ahh, ok. I got confused by your example code mentioning following.

```sql
ALTER TABLE "bars" ADD "foo_id" bigint, ADD CONSTRAINT "fk_rails_9e5a99443d" FOREIGN KEY ("foo_id") REFERENCES "foos" ("id");
```